rafthttp/transport.go

```diff
@@ -202,11 +202,16 @@ func (t *Transport) Stop() {
 	if tr, ok := t.pipelineRt.(*http.Transport); ok {
 		tr.CloseIdleConnections()
 	}
+	t.peers = nil
+	t.remotes = nil
 }
 
 func (t *Transport) AddRemote(id types.ID, us []string) {
 	t.mu.Lock()
 	defer t.mu.Unlock()
+	if t.remotes == nil {
+		panic("transport stopped")
```

Comment:
It seems to me the transport should really only be stopped once the http server is taken down. If `AddRemote` gets called after a `Stop`, then the listener is still active and the shutdown order is being ignored.